### ISSUE TITLE
How about a `redundant_ok` lint?

### ISSUE THREAD

**Issue by nathanielmanistaatgoogle** [rust-lang/rust-clippy#17271]
### What it does

I've noticed [a few places in the code with which I work](https://fuchsia.googlesource.com/fuchsia/+/a6449debce0f67f75c329bda150987ab7a3d15f3%5E%21/#F0) where `Ok(` and `?)` can be removed from the ends of a line without altering system behavior - would it be good to have Clippy flag those?

### Advantage

The advantage would be that the code on the page gets to avoid (a small bit of) unnecessary syntactic structural complexity.

### Drawbacks

_Maybe_ some folks like seeing the `Ok` as a visual reminder or cue of "this is an exit point of the current behavior"? Since there's not a `return` keyword like in other languages? ... but I'm just supposing and speculating about that.

### Example

```rust
Ok(helper_function_call(4)?)
```

Could be written as:

```rust
helper_function_call(4)
```

### Comparison with existing lints

I had figured that this probably already existed, but it doesn't yet seem to, nor did I find any past issue proposing it.

### Additional Context

I'm not aware of any additional context; feel free to inform me about what I am overlooking.

**Comment by nathanielmanistaatgoogle** [comment]
Wait, wait... I might have found this [with the name "needless"](https://rust-lang.github.io/rust-clippy/master/index.html#needless_question_mark) rather than "redundant"...

**Comment by CommanderStorm** [comment]
Sometimes, I think this is done to ".from" the error type if you don't quite care.
At least that is how I did this in the past.

Likely not the best pattern though, but one where I think this being a `restriction` lint might make sense.

(Not a maintainer and not a strong oppinion. Just trying to softly argue why people might do this)
